tests/models/test_truncated_bptt.py

```diff
@@ -14,82 +14,97 @@
 
 import pytest
 import torch
+from torch.utils.data import DataLoader, Dataset
 
 from pytorch_lightning import Trainer
 from tests.helpers import BoringModel
 
 
+class LinearModel(BoringModel):
+    """Linear model for testing TBPTT with automatic optimization."""
+
+    def __init__(self, truncated_bptt_steps=2, n_hidden_states=1, sequence_size=30, batch_size=30):
+        super().__init__()
+        self.truncated_bptt_steps = truncated_bptt_steps
+        self.n_hidden_states = n_hidden_states
+        self.sequence_size = sequence_size
+        self.batch_size = batch_size
+        self.automatic_optimization = True
+
+        self.example_input_array = torch.randn(5, truncated_bptt_steps)
+        self.layer = torch.nn.Linear(in_features=truncated_bptt_steps, out_features=truncated_bptt_steps)
+        self.test_hidden = None
+
+    def training_step(self, batch, batch_idx, hiddens):
+        assert hiddens == self.test_hidden, "Hidden state not persistent between tbptt steps"
```

Comment:
Can you add this hiddens assertion to the manual optimization model?

Reply:
it's called by the super().training_step already :)

Reply:
Oops, missed it!